src/com/mie/dao/PostDao.java

```diff
@@ -0,0 +1,188 @@
+			preparedStatement.setInt(1, postId);
+			preparedStatement.executeUpdate();
+
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	public void updatePost(Post post) {
+		try {
+			PreparedStatement preparedStatement = connection
+					.prepareStatement("update Posts set PostTime=?, UserId=?, GroupID=?, PostContent=?, LinkToPost=?"
+							+ " where PostID=?");
+			preparedStatement.setTimestamp(1, post.getPostedTime());
+			preparedStatement.setInt(2, post.getAuthorId());
+			preparedStatement.setInt(3, post.getGroupId());
+			preparedStatement.setString(4, post.getContent());
+			preparedStatement.setString(5, post.getUrl());
+			preparedStatement.setInt(6, post.getPostId());
+			preparedStatement.executeUpdate();
+
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	public List<Post> getAllPosts() {
+		List<Post> posts = new ArrayList<Post>();
+		try {
+			Statement statement = connection.createStatement();
+			ResultSet rs = statement.executeQuery("select * from Posts");
+			while (rs.next()) {
+				Post post = new Post();
+				post.setPostId(rs.getInt("PostID"));
+				post.setAuthorId(rs.getInt("UserID"));
+				post.setGroupId(rs.getInt("GroupID"));
+				post.setPostedTime(rs.getTimestamp("PostTime"));
+				post.setContent(rs.getString("PostContent"));
+				post.setUrl(rs.getString("LinkToPost"));
+				posts.add(post);
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+
+		return posts;
+	}
+	
+	public List<Post> getGroupPosts(Integer groupId) {
+		List<Post> posts = new ArrayList<Post>();
+		try {
+			Statement statement = connection.createStatement();
+			ResultSet rs = statement.executeQuery("select * from Posts where GroupID=" + groupId);
+			while (rs.next()) {
+				Post post = new Post();
+				post.setPostId(rs.getInt("PostID"));
+				post.setAuthorId(rs.getInt("UserID"));
+				post.setGroupId(rs.getInt("GroupID"));
+				post.setPostedTime(rs.getTimestamp("PostTime"));
+				post.setContent(rs.getString("PostContent"));
+				post.setUrl(rs.getString("LinkToPost"));
+				posts.add(post);
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+
+		return posts;
+	}
+	
+	public List<Post> getUSerPosts(Integer userId) {
+		List<Post> posts = new ArrayList<Post>();
+		try {
+			Statement statement = connection.createStatement();
+			ResultSet rs = statement.executeQuery("select * from Posts where UserID=" + userId);
+			while (rs.next()) {
+				Post post = new Post();
+				post.setPostId(rs.getInt("PostID"));
+				post.setAuthorId(rs.getInt("UserID"));
+				post.setGroupId(rs.getInt("GroupID"));
+				post.setPostedTime(rs.getTimestamp("PostTime"));
+				post.setContent(rs.getString("PostContent"));
+				post.setUrl(rs.getString("LinkToPost"));
+				posts.add(post);
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+
+		return posts;
+	}
+		
+	public Post getPost(Integer authorId, java.sql.Timestamp postTime, String content, String url) {
```

Comment:
Fix the Timestamp import I mentioned in a different comment and then the type for postTime can just be Timestamp